File: solutions/julia/pascals-triangle/1/pascals-triangle.jl
function binom(n,k)
    if n==0 || k==0 || k==n
        1
    else
        binom(n-1,k) + binom(n-1, k-1)
    end
end


function triangle(n)
    if n<0 
        throw(DomainError(n))
    else
        p_triangle = []
        for N in 0:(n-1) 
            row = []
            for k in 0:N 
                push!(row, binom(N,k))
            end
            push!(p_triangle,row)
        end
        p_triangle
    end
end

File: solutions/go/pascals-triangle/1/pascals_triangle.go
package pascal

func binom(n int, k int) int {
    if n==0 || k==0 || k==n {
        return 1
    } else {
        return binom(n-1,k)+binom(n-1,k-1)
    }
}

func Triangle(n int) [][]int {
	var p_triangle [][]int
    for N:=0; N<=n-1; N++ {
        var row []int
    	for k:=0;k<=N;k++ {
        	row = append(row, binom(N,k))
    	} 
    	p_triangle = append(p_triangle, row)
    }
    return p_triangle
}
